igvc_rviz_plugins/src/WaypointPanel.cpp

```diff
@@ -0,0 +1,61 @@
+#define _USE_MATH_DEFINES
+
+#include <igvc_rviz_plugins/WaypointPanel.h>
+#include <QVBoxLayout>
+#include <pluginlib/class_list_macros.h>
+#include <tf/LinearMath/Matrix3x3.h>
+
+namespace igvc_rviz_plugins {
+
+WaypointPanel::WaypointPanel(QWidget *parent)
+  : rviz::Panel(parent)
+{
+    ros::NodeHandle handle;
+
+    QLabel *label = new QLabel("0");
+
+    waypoint_subscriber = handle.subscribe<geometry_msgs::PointStamped>("/waypoint", 1, boost::bind(&WaypointPanel::waypoint_callback, this, _1, label));
+    robot_position_subscriber = handle.subscribe<nav_msgs::Odometry>("/odometry/filtered", 1, boost::bind(&WaypointPanel::robot_position_callback, this, _1, label));
+
+    QVBoxLayout *layout = new QVBoxLayout;
+    layout->addWidget(label);
+    setLayout(layout);
+}
+
+void WaypointPanel::waypoint_callback(const geometry_msgs::PointStampedConstPtr &msg, QLabel *label) {
+    way_x = msg->point.x - robot_x;
+    way_y = msg->point.y - robot_y;
+}
+
+void WaypointPanel::robot_position_callback(const nav_msgs::OdometryConstPtr &msg, QLabel *label) {
+    double distance = std::hypot(way_x, way_y);
+    double angle = atan2(way_y, way_x) * 180 / M_PI;
+
+    robot_x = msg->pose.pose.position.x;
+    robot_y = msg->pose.pose.position.y;
+
+    double qx = msg->pose.pose.orientation.x;
+    double qy = msg->pose.pose.orientation.y;
+    double qz = msg->pose.pose.orientation.z;
+    double qw = msg->pose.pose.orientation.w;
+
+    tf::Quaternion q(qx, qy, qz, qw);
+    tf::Matrix3x3 m(q);
+    double roll, pitch, yaw;
+    m.getRPY(roll, pitch, yaw);
+
+
+    //double yaw = atan2(siny, cosy) * 180 / M_PI;
```

Comment:
Try to avoid adding "dead code." (Commented out code). It's a good habit to build early. I've seen instances where huge long blocks of code are commented out so that another huge long alternative block of code below can be used instead.